compiler-core/src/javascript/pattern.rs

```diff
@@ -470,78 +495,111 @@ impl<'module_ctx, 'expression_gen, 'a> Generator<'module_ctx, 'expression_gen, '
 
                 let mut offset = Offset::new();
                 for segment in segments {
-                    match segment.options.as_slice() {
-                        [] | [Opt::Int { .. }] => {
-                            self.push_byte_at(offset.bytes);
-                            self.traverse_pattern(subject, &segment.value)?;
-                            self.pop();
-                            offset.increment(1);
-                            Ok(())
+                    if segment.type_ == crate::type_::int()
```

Comment:
Same here as above with doing it in two passes

Reply:
Done as per previous comment